GliaWidgets/Public/Glia/Glia+StartEngagement.swift

```diff
@@ -83,22 +83,24 @@ extension Glia {
         features: Features = .all,
         sceneProvider: SceneProvider? = nil
     ) throws {
-        try configure(
-            with: configuration,
-            queueId: queueID,
-            visitorContext: visitorContext
-        ) { [weak self] in
+        let completion = { [weak self] in
             self?.setChatMessageRenderer(messageRenderer: .webRenderer)
-            do {
-                try self?.startEngagement(
-                    engagementKind: engagementKind,
-                    theme: theme,
-                    features: features,
-                    sceneProvider: sceneProvider
-                )
-            } catch {
-                print("Engagement has not been started. Error='\(error)'.")
+            try self?.startEngagement(
+                engagementKind: engagementKind,
+                theme: theme,
+                features: features,
+                sceneProvider: sceneProvider
+            )
+        }
+        do {
+            try configure(
+                with: configuration,
+                queueId: queueID
+            ) {
+                try? completion()
```

Comment:
If this completion fails, what happens? I think it would fail silently, no? Maybe it would be good to also notify the integrator about the error in starting an engagement.

Reply:
I agree with Gerson that we should notify the integrator about an error

Reply:
In this case:
- it's not needed, because `startEngagement` throws an error when SDK is not configured (in this case SDK is already configured) or active engagement exists (in this case `configure` will throw `GliaError.configuringDuringEngagementIsNotAllowed` and `try completion()` on 103th line will be executed.
- I have not found any way to throw an error from `configure` completion scope, because this completion closure is not throwable (i.e. not `throws`)